### ISSUE TITLE
Not needed span element in Translate component

### ISSUE THREAD

**Issue by skrywus** [JSxMachina/react-i18nify#16]
Please consider removing span element from Translate component. Many times it breaks styling and be a reason on many problems.

**Comment by karpoff** [comment]
you can use I18n.t
var I18n = require('react-i18nify').I18n;
I18n.t('application.title');

see Helpers section of readme

**Comment by skrywus** [comment]
"When the translation object or locale values are set, all instances of Translate and Localize will be re-rendered to reflect the latest state. If you choose to use I18n.l or I18n.t then it up to you to handle state change."
I need to use Translate in my case, because it can re-render itself as you see in the description above.

**Comment by gerbenmeyer** [comment]
The span is required to make Translate (and Localize) valid React elements, so it cannot be removed. With respect to re-rendering placeholders and so, I'm open to suggestions on how to tackle that. There's a separate issue already for that: https://github.com/zoover/react-i18nify/issues/13

**Comment by flo-sch** [comment]
Hey @gerbenmeyer, sorry to comment on a closed issue, it but I might have a suggestion for this problem.

Some libraries such as reactstrap use a `tag` prop, which could silently default to a `<span>` to preserve the current behaviour.

Basically something like :

```
static propTypes = {
  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
};

static defaultProps = {
  tag: 'span'
};

render() {
    const {
      tag, value, dangerousHTML, style, className,
    } = this.props;
    const translation = I18n._translate(value, this.otherProps());

    if (dangerousHTML) {
      return (
        <tag
          style={style}
          className={className}
          dangerouslySetInnerHTML={{ __html: translation }}
        />
      );
    }
    return <tag style={style} className={className}>{translation}</tag>;
}
```

Which could be used as following :

```
<Translate value="my.translated.key" />
---> <span>my translated value</span>

<Translate tag="div" value="my.translated.key" />
---> <div>my translated value</div>

<Translate tag="h1" value="my.translated.key" />
---> <h1>my translated value</h1>
```

What do you think?

I can propose a PR for this if you are interested.

**Comment by gerbenmeyer** [comment]
This is actually a good idea!

Another thing I was thinking of was to directly return the translation, as with react 16, returning strings from a render function is also possible. Only doing this though would not be backward compatible.

So your solution would be even more awesome with the addition that if you set `tag` to `null`, no html tag is rendered at all, but the translation is directly returned.

**Comment by flo-sch** [comment]
Glad you like it, I start working on a PR right now :)

**Comment by flo-sch** [comment]
@gerbenmeyer PR (#40) on its way :)

Just couldn't figure out how to allow a `null` value with PropTypes ;

Do you have any idea about that point?

**Comment by gerbenmeyer** [comment]
Nice work, merged it!

I have no idea exactly about the `null`, I'll experiment a bit as well :smile:

**Comment by flo-sch** [comment]
Alright, thank you very much :)

Do you think that could be released in the coming days ? (1.9.2... ?)

**Comment by gerbenmeyer** [comment]
Already did! (1.10.0)
